### DISCUSSION TITLE
Validate on initial render

### DISCUSSION THREAD

**Discussion by thaorell** [react-hook-form/react-hook-form#7702] [General] [answered]
Hi,

my form (mode onBlur) has default values that are not valid. For [example](https://codesandbox.io/s/react-hook-form-with-ui-library-forked-kyqdl?file=/src/index.js), the `firstName` default value is '', which is not validate under the validation schema, and therefore I want to gray out the `Submit` button on render. Is there a way to do that?

**Comment by Moshyfawn** [comment, marked as answer]
Hello!

First of all, [your CSB](https://codesandbox.io/s/react-hook-form-with-ui-library-forked-kyqdl?file=/src/index.js) has the **beta** release of RHF version 7 installed, which isn't recommended to use.
![image](https://user-images.githubusercontent.com/16290753/151866824-f87f775f-08fe-4ef2-af31-a65e617c8490.png)

To check an empty string input as an invalid initial value, you should use `required` attribute instead of `minLength`. See [this CSB](https://codesandbox.io/s/react-hook-form-with-ui-library-forked-ly9zs?file=/src/index.js) with a working example

**Reply by Moshyfawn** [reply]
For more info you can check [the HTML specs](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#setting-minimum-input-length-requirements:-the-minlength-attribute) on `minLength` attribute

**Reply by thaorell** [reply]
I completely overlooked that. Thanks
